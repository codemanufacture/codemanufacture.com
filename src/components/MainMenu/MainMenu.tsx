import * as React from 'react'
import styled from 'styled-components'
import classNames from 'classnames'
import { colors, transitions, displayDimensions, sizes } from '../../theme'
import MenuItem from '../MenuItem'

const StyledMenu = styled.ul`
  display: flex;
  margin: 0;
  padding: 0;
  will-change: auto;

  li {
    list-style: none;
  }

  a {
    display: block;
    padding: 24px 22px;
    font-size: 12px;
    color: ${colors.link};
    text-decoration: none;
    text-transform: uppercase;
    letter-spacing: 3px;
    transition: color ${transitions.basicTransition};
  }

  &:hover,
  &.active {
    color: ${colors.linkActive};
  }

  @media (max-width: ${displayDimensions.tabletSize}) {
    width: 100%;
    top: ${sizes.headerHeight};
    position: absolute;
    transform: translateX(100%);
    flex-direction: column;
    background: ${colors.background};
    transition: transform ${transitions.basicTransition};
    border-bottom: 1px solid ${colors.borderColor};

    &.active {
      right: 0;
      transform: translateX(0);
    }

    li {
      text-align: center;
    }
  }
`

interface MainMenuProps {
  isMobileMenuActive: boolean
  onMenuClick(): void
}

const MainMenu: React.FunctionComponent<MainMenuProps> = ({
  isMobileMenuActive,
  onMenuClick,
}) => {
  const menuClassName = classNames({
    active: isMobileMenuActive,
  })

  return (
    <StyledMenu onClick={onMenuClick} className={menuClassName}>
      <MenuItem pagePath="/homepage#home" name="Home" />
      <MenuItem pagePath="/homepage#services" name="Services" />
    </StyledMenu>
  )
}

export default MainMenu
