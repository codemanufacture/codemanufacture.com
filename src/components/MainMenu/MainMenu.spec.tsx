import * as React from 'react'
import renderer from 'react-test-renderer'
import MainMenu from './MainMenu'

describe('MainMenu component', () =>
  it('renders correctly', () => {
    const props = {
      isMobileMenuActive: false,
      onMenuClick: jest.fn(),
    }

    const tree = renderer.create(<MainMenu {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
