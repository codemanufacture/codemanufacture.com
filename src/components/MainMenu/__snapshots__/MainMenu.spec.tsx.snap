// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu component renders correctly 1`] = `
<ul
  className="sc-bdVaJa kKdKPP"
  onClick={[MockFunction]}
>
  <li>
    <a
      href="/homepage#home"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="Home"
    >
      Home
    </a>
  </li>
  <li>
    <a
      href="/homepage#services"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="Services"
    >
      Services
    </a>
  </li>
  <li>
    <a
      href="/homepage/#contact"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="Contact"
    >
      Contact
    </a>
  </li>
</ul>
`;
